internal/security/secretstore/kongadminapi.go

```diff
@@ -0,0 +1,241 @@
+	if err != nil {
+		return "", fmt.Errorf("%s Could not parse JWT duration: %w", k.prefixes.errText, err)
+	}
+
+	// Sanity check - parse & check EC key
+	eckey, err := jwt.ParseECPrivateKeyFromPEM([]byte(privateKey))
+	if err == nil && eckey.Params().BitSize != 256 {
+		return "", fmt.Errorf("%s EC key bit size is incorrect (%d instead of 256)", k.prefixes.errText, eckey.Params().BitSize)
+	}
+	if err != nil {
+		return "", fmt.Errorf("%s Could not parse private key: %w", k.prefixes.errText, err)
+	}
+
+	// Create JWT
+	token := jwt.NewWithClaims(jwt.SigningMethodES256, &jwt.StandardClaims{
+		Issuer:    issuer,
+		IssuedAt:  now,
+		NotBefore: now,
+		ExpiresAt: now + int64(duration.Seconds()),
+	})
+
+	// Save JWT to string return
+	signedToken, err = token.SignedString(eckey)
+	if err != nil {
+		return "", fmt.Errorf("%s Could not sign JWT: %w", k.prefixes.errText, err)
+	}
+
+	return
+}
+
+// clearSecrets clears the key structs by setting them to empty struct values
+func (k *KongAdminAPI) clearSecrets() {
+	k.secrets = KongAdminAPISecrets{}
+}
+
+// verifyPrefixes confirms that there are values set for the prefixes
+func (k *KongAdminAPI) verifyPrefixes() error {
+
+	base := "[build-kong-admin]"
+
+	if k.prefixes.cmdText == "" {
+		k.prefixes.cmdText = base
+	}
+
+	if k.prefixes.errText == "" {
+		k.prefixes.errText = base + "[error]"
+	}
+
+	if k.prefixes.infoText == "" {
+		k.prefixes.infoText = base + "[info]"
+	}
+
+	return nil
+}
+
+// verifyPaths confirms values are set for the paths, and that those paths exist
+func (k *KongAdminAPI) verifyPaths() error {
+
+	// Is the template path set?
+	if k.paths.template == "" {
+		return fmt.Errorf("%s Template file path (<KongAdminAPI>.paths.template) is not set - check configuration.toml", k.prefixes.errText)
+	}
+
+	// Does the template path/file exist? It should be set.
+	if _, err := os.Stat(k.paths.template); os.IsNotExist(err) {
+		return fmt.Errorf("%s Template file path (<KongAdminAPI>.paths.template) does not exist - check configuration.toml", k.prefixes.errText)
+	}
+
+	// Is the config file path set?
+	if k.paths.config == "" {
+		return fmt.Errorf("%s Kong configuration file path (<KongAdminAPI>.paths.config) is not set - check configuration.toml", k.prefixes.errText)
+	}
+
+	// Does the config file path directory exist? If not, create it.
+	if _, err := os.Stat(path.Dir(k.paths.config)); os.IsNotExist(err) {
+		dirErr := os.MkdirAll(path.Dir(k.paths.config), 0777)
+		if dirErr != nil {
+			return fmt.Errorf("%s Kong configuration file path (<KongAdminAPI>.paths.config) could not be created", k.prefixes.errText)
+		}
+	}
+
+	// Is the JWT path set?
+	if k.paths.jwt == "" {
+		return fmt.Errorf("%s JWT file path (<KongAdminAPI>.paths.jwt) is not set - check configuration.toml", k.prefixes.errText)
+	}
+
+	// Does the JWT file path directory exist? If not, create it.
+	if _, err := os.Stat(k.paths.jwt); os.IsNotExist(err) {
+		dirErr := os.MkdirAll(path.Dir(k.paths.jwt), 0777)
```

Comment:
Here as well.

Reply:
This has to be world read since it's created by the "edgex_user" user and then traversed in the Kong process by the "kong" user.

Reply:
Ok - I'm actually updating the permissions to "644" on this directory and I'm going to give it a test. If the current user has read/write and everyone else has read...I think that will be sufficient.

Reply:
Directories need the execute bit set. That means 700, 750, 755.

Reply:
Ok - that worked. Is that permission level sufficient?